### DISCUSSION TITLE
git pull reported uncommited changes? No local changes made

### DISCUSSION THREAD

**Discussion by creeduk** [Mikubill/sd-webui-controlnet#1227] [Q&A]
When I first installed controlnet1.1 I archived 1.0, created an empty folder and used git pull. This worked fine.

Today I tried to update with git pull. I had A1111 offline so rather than use the update module and because I manually installed I decided to use pull/

I had an error regarding gitignore and README.md. I never touched them. I checked the modified and creation date and it matched exactly the other files. 

Is it possible when I pulled they were not committed properly or something?

```
C:\Users\sd_test\stable-diffusion-webui\extensions\sd-webui-controlnet>git pull
remote: Enumerating objects: 883, done.
remote: Counting objects: 100% (491/491), done.
remote: Compressing objects: 100% (91/91), done.
remote: Total 883 (delta 418), reused 422 (delta 400), pack-reused 392
 MiB/s
Receiving objects: 100% (883/883), 1.70 MiB | 3.04 MiB/s, done.
Resolving deltas: 100% (533/533), completed with 32 local objects.
From https://github.com/Mikubill/sd-webui-controlnet
   2bc4400..c9c8ca6  main       -> origin/main
   67c40c7..c31d7d1  lvmin      -> origin/lvmin
error: Your local changes to the following files would be overwritten by merge:
        .gitignore
        README.md
Please commit your changes or stash them before you merge.
Aborting
Updating 2bc4400..c9c8ca6
```

Any way I did not want this to be an issue. I considered deleting and just installing fresh but decided to just try git reset --hard HEAD:

```
C:\Users\sd_test\stable-diffusion-webui\extensions\sd-webui-controlnet>git reset --hard HEAD
HEAD is now at 2bc4400 Merge pull request #1069 from Mikubill/lvmin
```

this seems to have worked I was able to git pull now. 

I am not very familiar with github, ca anybody explain why it had this issue if I did not edit the files? I kind of wish I backed that folder up so I could try and compare my local README to the online from that date.

**Comment by lllyasviel** [comment]
we removed obsolete branches. not sure if that will help

**Reply by creeduk** [reply]
Maybe that was it. Seems to update and work OK now must have just pulled at a bad time :)
